src/Task4/ListOfNumbers.java

```diff
@@ -0,0 +1,40 @@
+package Task4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListOfNumbers {
+    private final List<Double> doubleList = new ArrayList<>();
+
+    public void addNumber(Double skaicius){
+        doubleList.add(skaicius);
+    }
+
+    public double getAverage(){
+        Double sum = 0.00;
+        for (Double d:doubleList){
+            sum+=d;
+        }
+        return sum/doubleList.size();
+    }
+
+    public static ListOfNumbers findWithMaxAverage(ListOfNumbers...listObject) {
+        double maxAverage = 0.00;
+        ListOfNumbers withMaxAverage = new ListOfNumbers();
```

Comment:
šitą geriau daryti null, o ne new ListOfNumbers() - nes nepadavus jokio listObject į funkciją - jis tau sakys "radau ListOfObject su didžiausiu vidurkiu - tai šitas tuščias "new ListOfObject()""